server/src/main/java/org/elasticsearch/cluster/routing/RecoverySource.java

```diff
@@ -319,4 +325,39 @@ public boolean expectEmptyRetentionLeases() {
             return false;
         }
     }
+
+    /**
+     * Recovery of a shard that is created as a result of a resharding split.
+     * Not to be confused with _split API.
+     */
+    public static class ReshardSplitTargetRecoverySource extends RecoverySource {
```

Comment:
I may be worrying too much about naming but i don't want to call it `Split` because of `_split` API and i want `Target` in there to disambiguate. So i came up with this.